source/leop.c

```diff
@@ -48,61 +48,71 @@ bool hard_switch_status() {
     TickType_t twenty_sec_delay = pdMS_TO_TICKS(20 * 1000);
     int getStatus_retries;
     int successful_deployment = 0;
+    uint16_t burnwire_currents[7] = {0};
+
     //sw = {Port, UHF_P, UHF_Z, Payload, UHF_S, UHF_N, Starboard, DFGM};
     for (getStatus_retries = 0; getStatus_retries <= MAX_RETRIES; getStatus_retries++) {
         sw = 0;
         //Deploy DFGM
-        if (switchstatus(sw) != 1 && getStatus_retries != MAX_RETRIES) {
-            //ex2_log("Check #%d: %c not deployed\n", &getStatus_retries, sw);           
-            //ex2_log("Manually activated %c\n", sw);
-            activate(sw);
-            //vTaskDelay(twenty_sec_delay);
+        if ((switchstatus(sw) != 1) && (getStatus_retries != MAX_RETRIES)) {
+            ex2_log("Check #%d: %c not deployed\n", &getStatus_retries, sw);
+            ex2_log("Activated %c\n", sw);
+            activate(sw, &burnwire_currents[sw]);
+            vTaskDelay(5000);
         }
-        else if (getStatus_retries == MAX_RETRIES) {
-            //ex2_log("Check #%d: %c not deployed, exiting the LEOP sequence.\n", &getStatus_retries, sw);
+        else if ((switchstatus(sw) != 1) && (getStatus_retries == MAX_RETRIES)) {
+            ex2_log("Check #%d: %c not deployed, exiting the LEOP sequence.\n", &getStatus_retries, sw);
             return false;
         }
     }
-    //vTaskDelay(two_min_delay);
+    ex2_log("DFGM deployed, burnwire current = %d\n", burnwire_currents[0]);
+    vTaskDelay(5000);
+    //    vTaskDelay(two_min_delay);
```

Comment:
as a general comment, we should finalize the delays required between each deployment

Reply:
Yeah I'm going to revert to the full periods